### PR TITLE
Vendor warehouse

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java [hunk 1]
```diff
@@ -111,23 +137,39 @@ private WarehouseId findPickingWarehouseId(@NonNull final I_C_Order order)
...
 		}
 
-		// if order is a purchase order, return null
 		return null;
 	}
 
+	@Nullable
+	private WarehouseId findWarehouseIdForSOTrx(@NonNull final BPartnerId bpartnerId, @NonNull final SOTrx soTrx)
+	{
+		final IBPartnerDAO partnerDAO = Services.get(IBPartnerDAO.class);
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java:152]: why not making `partnerDAO` a field?

File: backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java [hunk 2]
```diff
@@ -111,23 +137,39 @@ private WarehouseId findPickingWarehouseId(@NonNull final I_C_Order order)
...
+			{
+				return null;
+			}
+			return WarehouseId.ofRepoIdOrNull(bp.getM_Warehouse_ID());
+		}
+
+		if (!bp.isVendor())
+		{
+			return null;
+		}
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java:167]: i find this logic a bit hard to read
why not writing like
```java
if (soTrx.isSales()) && bp.isCustomer()) { return ... }
else if (soTrx.isPurchase() && bp.isVendor()) { return ... }
```

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.business/src/main/java/org/adempiere/warehouse/spi/impl/WarehouseAdvisor.java:163]: Done

**Pull request merged**